src/main/java/com/checkmarx/flow/custom/GitHubIssueTracker.java

```diff
@@ -185,7 +188,7 @@ public Issue updateIssue(Issue issue, ScanResults.XIssue resultIssue, ScanReques
         ResponseEntity<com.checkmarx.flow.dto.github.Issue> response;
         try {
             response = restTemplate.exchange(issue.getUrl(), HttpMethod.POST, httpEntity, com.checkmarx.flow.dto.github.Issue.class);
-            this.addComment(Objects.requireNonNull(response.getBody()).getUrl(),"Issue still exists. ");
+            addCommentToAnUpdatedIssue(Objects.requireNonNull(response.getBody()).getUrl(), createNewIssueStatus(issue, resultIssue, response.getBody()));
```

Comment:
What about extracting the commenting logic into a separate class, e.g. CommentFormatter?